In [1]:
import numpy as np

# Empty and Full Numpy Array
#### Create a 3x4 empty numpy array
#### Create a 3x4 full numpy array where every element is equal to 55.

In [6]:
empty_array = np.empty((3, 4), dtype=int)
print(empty_array)

[[94623256455168              0   206158430253   193273528376]
 [  227633266738   210453397588   249108103218   223338299442]
 [  219043332154   197568495670   236223201332   386547056692]]


In [7]:
full_array = np.full((3, 4), 55, dtype=float)
print(full_array)

[[55. 55. 55. 55.]
 [55. 55. 55. 55.]
 [55. 55. 55. 55.]]


# Find Numpy Non-Zero Mean
Find the average of non-zero elements of a = np.array([1, 0, 2, 0, 3, 0, 0, 5, 6, 7, 5, 0, 56]) .

In [30]:
a = np.array([1, 0, 2, 0, 3, 0, 0, 5, 6, 7, 5, 0, 56])

In [31]:
a[a != 0].mean()

10.625

# Convert Python Sequences to 3D ndarray
Create a nested list in Python and store it in the python_list3 variable. Then create a three-dimensional ndarray using the python_list3 variable and store it in the ndarray_3d variable.

In [20]:
python_list3 = [[[1, 2, 3], [2, 3, 5], [3, 4, 6]], [[1, 2, 3], [2, 3, 5], [3, 4, 6]], [[1, 2, 3], [2, 3, 5], [3, 4, 6]]]

In [22]:
ndarray_3d = np.array(python_list3)
ndarray_3d

array([[[1, 2, 3],
        [2, 3, 5],
        [3, 4, 6]],

       [[1, 2, 3],
        [2, 3, 5],
        [3, 4, 6]],

       [[1, 2, 3],
        [2, 3, 5],
        [3, 4, 6]]])

# An array with 4-byte elements
Create a 2-dimensional array of size 3 x 4 named x, composed of 4-byte integer elements:

In [34]:
np.full((3, 4) , 6, dtype='i4')

array([[6, 6, 6, 6],
       [6, 6, 6, 6],
       [6, 6, 6, 6]], dtype=int32)

# Evenly Spaced Values Numpy Array
Create an array containing the numbers 0 to 999 and store it in variable a1 .

In [2]:
a1 = np.arange(0, 1000)

In [3]:
a1

array([  0,   1,   2,   3,   4,   5,   6,   7,   8,   9,  10,  11,  12,
        13,  14,  15,  16,  17,  18,  19,  20,  21,  22,  23,  24,  25,
        26,  27,  28,  29,  30,  31,  32,  33,  34,  35,  36,  37,  38,
        39,  40,  41,  42,  43,  44,  45,  46,  47,  48,  49,  50,  51,
        52,  53,  54,  55,  56,  57,  58,  59,  60,  61,  62,  63,  64,
        65,  66,  67,  68,  69,  70,  71,  72,  73,  74,  75,  76,  77,
        78,  79,  80,  81,  82,  83,  84,  85,  86,  87,  88,  89,  90,
        91,  92,  93,  94,  95,  96,  97,  98,  99, 100, 101, 102, 103,
       104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116,
       117, 118, 119, 120, 121, 122, 123, 124, 125, 126, 127, 128, 129,
       130, 131, 132, 133, 134, 135, 136, 137, 138, 139, 140, 141, 142,
       143, 144, 145, 146, 147, 148, 149, 150, 151, 152, 153, 154, 155,
       156, 157, 158, 159, 160, 161, 162, 163, 164, 165, 166, 167, 168,
       169, 170, 171, 172, 173, 174, 175, 176, 177, 178, 179, 18

# End of the interval in ndarray in a range of numbers
Create an array of dtype float containing the numbers -3 to 3 and store it in variable a2 .

(The number 3 itself is not in the array)

In [6]:
a2 = np.arange(-3, 3, dtype='f4')

In [8]:
a2

array([-3., -2., -1.,  0.,  1.,  2.], dtype=float32)

# Ants move in the numpy array
Create an array that starts at 2 and goes up to 10 in steps of 0.1 and store it in variable a3

(The number 10 itself is not in the array)

In [9]:
a3 = np.arange(2, 10, 0.1, dtype=float)

In [10]:
a3

array([2. , 2.1, 2.2, 2.3, 2.4, 2.5, 2.6, 2.7, 2.8, 2.9, 3. , 3.1, 3.2,
       3.3, 3.4, 3.5, 3.6, 3.7, 3.8, 3.9, 4. , 4.1, 4.2, 4.3, 4.4, 4.5,
       4.6, 4.7, 4.8, 4.9, 5. , 5.1, 5.2, 5.3, 5.4, 5.5, 5.6, 5.7, 5.8,
       5.9, 6. , 6.1, 6.2, 6.3, 6.4, 6.5, 6.6, 6.7, 6.8, 6.9, 7. , 7.1,
       7.2, 7.3, 7.4, 7.5, 7.6, 7.7, 7.8, 7.9, 8. , 8.1, 8.2, 8.3, 8.4,
       8.5, 8.6, 8.7, 8.8, 8.9, 9. , 9.1, 9.2, 9.3, 9.4, 9.5, 9.6, 9.7,
       9.8, 9.9])

# Uniformly Spaced Numbers
Create an array containing 5 elements between the numbers 1 to 122 in such a way that the numbers 1 and 121 are part of this array and the distance between the numbers should be equal. store the array in a5 variable

In [15]:
a5 = np.linspace(1, 125, num=5)

In [16]:
a5

array([  1.,  32.,  63.,  94., 125.])